application.cpp

```diff
@@ -128,23 +133,24 @@ void application::startup() {
 
    //after startup, shut down the signal handling thread and catch the signals back on main io_service
    clean_up_signal_thread();
-   setup_signal_handling_on_ios(get_io_service());
+   setup_signal_handling_on_ios(get_io_service(), false);
+
+   std::shared_ptr<boost::asio::signal_set> sighup_set(new boost::asio::signal_set(*io_serv, SIGHUP));
```

Comment:
These two lines should also be guarded by `#ifdef SIGHUP`.

Reply:
Also you could use `get_io_service()` instead of `*io_serv`.

Reply:
Done.